### PR TITLE
gitinterface: Support verifying SSH key signatures

### PR DESCRIPTION
**Pull request by adityasaky**
This PR adds support for verifying Git signatures issued using SSH keys (we need to evaluate the missing algorithms / key types still). This goes hand in hand with https://github.com/secure-systems-lab/go-securesystemslib/pull/67 and is related to #229.

### CODE REVIEW

**Comment by adityasaky** [comment]
https://github.com/secure-systems-lab/go-securesystemslib/pull/67 now blocks this PR. I've added test code from that PR here that must be taken out.

I've also added ed25519 keys but there's an inconsistency in supported formats between `ssh-keygen` and securesystemslib, though I haven't dived into python-securesystemslib for this specifically yet. I've added test for commits signed using RSA and ECDSA keys generated using ssh-keygen.

**Comment by adityasaky** [comment]
@wlynch I think this is ready for review. The ed25519 thing hasn't been resolved yet but if everything else looks fine, I'm okay to proceed with a ticket for that one. I think it might primarily be documentation related that talks about what formats to use / how to convert between them, with example commands.

File: go.mod [hunk 1]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/go-git/go-git/v5 v5.11.0
 	github.com/in-toto/attestation v1.0.1
 	github.com/jonboulle/clockwork v0.4.0
-	github.com/secure-systems-lab/go-securesystemslib v0.8.0
+	github.com/secure-systems-lab/go-securesystemslib v0.8.1-0.20240108171218-da429971be5a
```
Comment (adityasaky) [inline go.mod:11]: Also, this has been merged into main. dependabot will auto-update on the next release, v0.9.0.

File: internal/gitinterface/keys.go [hunk 2]
```diff
@@ -257,3 +268,66 @@ func verifyGitsignSignature(ctx context.Context, key *tuf.Key, data, signature [
...
+	sshSignature, err := decodeSSHSignature(signature)
+	if err != nil {
+		return errors.Join(ErrVerifyingSSHSignature, err)
+	}
+
+	if err := publicKey.Verify(data, sshSignature); err == nil {
+		return nil
+	}
+
+	return ErrIncorrectVerificationKey
```
Comment (wlynch) [inline internal/gitinterface/keys.go:293]:
```suggestion
	if err := publicKey.Verify(data, sshSignature); err != nil {
		return errors.Join(ErrIncorrectVerificationKey, err)
	}
	return nil
```

we probably want to join the errors here too, this way we don't drop any debug information about why the verify may have failed.

**Review by wlynch — CHANGES_REQUESTED** [review]
Looks great! One small change.

File: internal/gitinterface/keys.go (hunk 2 above)
Comment (adityasaky) [reply internal/gitinterface/keys.go]: Done! Took the liberty of updating this in verifyGitsignSignature as well.

**Review by wlynch — APPROVED** [review]

**Pull request merged**